crates/prompt_store/src/prompts.rs

```diff
@@ -124,10 +138,43 @@ impl PromptBuilder {
         .unwrap_or_else(|| Arc::new(Self::new(None).unwrap()))
     }
 
+    /// Helper function for handlebars templates to check if a specific tool is enabled
+    fn has_tool_helper(
+        h: &handlebars::Helper,
+        _: &Handlebars,
+        ctx: &handlebars::Context,
+        _: &mut handlebars::RenderContext,
+        out: &mut dyn handlebars::Output,
+    ) -> handlebars::HelperResult {
+        let tool_name = h.param(0).and_then(|v| v.value().as_str()).ok_or_else(|| {
+            handlebars::RenderError::new("hasTool helper: missing or invalid tool name parameter")
+        })?;
+
+        let enabled_tools = ctx
+            .data()
+            .get("available_tools")
+            .and_then(|v| v.as_array())
+            .map(|arr| arr.iter().filter_map(|v| v.as_str()).collect::<Vec<&str>>())
+            .ok_or_else(|| {
+                handlebars::RenderError::new(
+                    "hasTool handlebars helper: available_tools not found or not an array",
+                )
+            })?;
+
+        if enabled_tools.contains(&tool_name) {
+            out.write("true")?;
+        }
+
+        Ok(())
+    }
+
     pub fn new(loading_params: Option<PromptLoadingParams>) -> Result<Self> {
         let mut handlebars = Handlebars::new();
         Self::register_built_in_templates(&mut handlebars)?;
 
+        // Register the hasTool helper to check if a tool is enabled
```

Comment:
This comment here doesn't seem necessary, as it is just repeating what is apparent in the code.